src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1799,6 +1799,16 @@ public override BoundNode VisitObjectCreationExpression(BoundObjectCreationExpre
             return null;
         }
 
+        public override BoundNode VisitUnconvertedObjectCreationExpression(BoundUnconvertedObjectCreationExpression node)
```

Comment:
Can this be removed or throw?

Reply:
There are places were we deliberately visit unconverted nodes 

- GetArgumentsForMethodTypeInference yields the unbound lambda for type reinference
- `VisitConvertedTupleLiteral` visits `SourceTuple` for semantic model

(there might be more)

Reply:
I expect there are more, do we have tests for what the SpeculativeSemanticModel will do when a `new()` is encountered?